framework/include/userobject/SolutionUserObject.h

```diff
@@ -75,6 +76,91 @@ class SolutionUserObject : public GeneralUserObject
   virtual Real pointValue(Real t, const Point & p, const unsigned int local_var_index) const;
 
   /**
+   * Returns a value at a specific location and variable (see SolutionFunction)
+   * @param t The time at which to extract (not used, it is handled automatically when reading the data)
+   * @param p The location at which to return a value
+   * @param var_name The variable to be evaluated
+   * @return The desired value for the given variable at a location
+   */
+  virtual Real pointValue(Real t, const Point & p, const std::string & var_name) const;
```

Comment:
You are overloading virtual functions. That is not a good idea. Once somebody inherits from this, it will hide the other method in the child class. Either you do not expect people to override and then you do not need `virtual` or you do, but then pick unique names... Similarly below with other methods.